### ISSUE TITLE
Can't interpolate in docker dns_servers field

### ISSUE THREAD

**Issue by kyhavlov** [hashicorp/nomad#1214]
### Nomad version

`Nomad v0.3.2`
### Issue

Nomad doesn't seem to interpolate variables in the dns_servers field; the list of DNS servers doesn't get provided to the container until runtime so Nomad should have access to them at that point. It also won't work with 'meta' fields or environment variables.

What I'm trying to do is pass the ip of the node so that the container can use the local consul agent for DNS. If there's another way to pass the address in dynamically that would work too, but I couldn't find one.
### Nomad Server logs (if appropriate)

`[ERR] driver.docker: invalid ip address for container dns server: ${attr.unique.network.ip-address}`
### Job file (if appropriate)

```
task "app-container" {
    driver = "docker"

    config {
        image = "hello-world"
        dns_servers = ["${attr.unique.network.ip-address}"]
    }
}
```

**Comment by spicykaiju** [comment]
Greetings @diptanu - any idea when this will be worked on?

**Comment by kyhavlov** [comment]
For posterity, the way I ended up doing this was to set "--dns=${NODE_IP}" on the args for the docker daemon.

**Comment by rokka-n** [comment]
@kyhavlov how did you configure consul/dnsmasq on the host?

**Comment by kyhavlov** [comment]
@rokka-n I had consul bind on the docker0 interface (`172.17.0.1`) and port 53, then gave `--dns=172.17.0.1` as an argument to the containers.

**Comment by rokka-n** [comment]
Thank you, all worked.
I bind dnsmasq to all interface.

**Comment by dadgar** [comment]
Fixed in 0.5.0: https://github.com/hashicorp/nomad/pull/1965
